core/types.ml

```diff
@@ -1995,28 +2009,197 @@ module Policy = struct
                 |> sync)
 
   let effect_sugar
-  = Settings.(flag "effect_sugar"
-              |> synopsis "Toggles the effect sugar in pretty printer."
-              |> convert parse_bool
-              |> sync)
+    = Settings.(flag "effect_sugar"
+                |> synopsis "Toggles the effect sugar in pretty printer."
+                |> convert parse_bool
+                |> sync)
 
-  type t = { quantifiers  : bool
-           ; flavours     : bool
-           ; hide_fresh   : bool
-           ; kinds        : kind_policy
-           ; effect_sugar : bool }
+  module EffectSugar : sig
+    type opt = PresenceOmit
+             | AliasOmit
+             (* vvv TODO name; for now it's the One Effect To Rule Them All *)
+             | ArrowsShowTheOneEffect
+             | ArrowsCurriedCollectionAssumeFresh
```

Comment:
How about CurriedCollectionFresh? It doesn't imply that a transformation will happen (as this setting is for now just about an assumption to apply when printing). I also wanted to keep it clear that this is about the arrows where arguments are just being collected (this may not be necessary though).

Reply:
The suggested name means absolutely nothing to me. It is not clear what it's semantics is.

Reply:
I'd settle for `ArrowsCurriedHideFresh`, this seems it should be clear enough. I'll update the PR description too.